node/test/balance_peer_requests.js

```diff
@@ -0,0 +1,126 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+var parallel = require('run-parallel');
+var MockTimers = require('time-mock');
+var allocCluster = require('./lib/alloc-cluster.js');
+
+allocCluster.test('requests are balanced evenly across peers', {
+    numPeers: 5,
+    timers: MockTimers(Date.now()),
+    channelOptions: {
+        random: winning
+    }
+}, function t(cluster, assert) {
+    var requestsPerPeer = 2;
+
+    cluster.tiberii = cluster.channels.map(function makeTiberius(channel) {
+        return channel.makeSubChannel({
+            serviceName: 'tiberius'
+        });
+    });
+
+    clique(cluster.tiberii, onCliqued);
+
+    function onCliqued(err, value) {
+        if (err) return assert.end(err);
+
+        setupServiceCluster(cluster);
+
+        var head = cluster.tiberii[0];
+        var tail = cluster.hosts.slice(1);
+
+        var jobs = [];
+        var responderCounts = {};
+        var requestCount = (cluster.channels.length - 1) * requestsPerPeer;
+        for (var index = 0; index < requestCount; index++) {
+            /*jshint -W083 */
+            jobs.push(function sendRequest(callback) {
+                head.request({
+                    serviceName: 'tiberius',
+                    timeout: 2000
+                }).send('slow', 'okay?', 'okay?', onResponse);
+                function onResponse(err, res, hostPort, result) {
+                    responderCounts[hostPort] = (responderCounts[hostPort] || 0) + 1;
+                    callback();
+                }
+            });
+            /*jshint +W083 */
+        }
+
+        parallel(jobs, onDone);
+
+        setTimeout(function () {
+            cluster.timers.advance(500);
+            setTimeout(function () {
+                cluster.timers.advance(Infinity);
+            }, 200);
+        }, 200);
+
+        function onDone(err) {
+            if (err) return assert.end(err);
+            for (var index = 0; index < tail.length; index++) {
+                var count = responderCounts[tail[index]];
+                assert.equal(count, requestsPerPeer, 'responder ' + index + ' received 2 requests');
+            }
+            assert.end();
+        }
+    }
+});
+
+
+function setupServiceCluster(cluster) {
+    cluster.tiberii.forEach(function (tiberius) {
+        tiberius.register('slow', function (req, res) {
+            cluster.timers.setTimeout(function delayedOk() {
+                res.sendOk(tiberius.hostPort, 'dokie');
+            }, 500);
+        });
+    });
+}
+
+function clique(channels, callback) {
```

Comment:
Hmm, I'm doing similar things recently for xlang testing to say "connect, wait for identified, and then ready"... we can probably pull this and my logic out into convenience on allocCluster directly, especially once it shifts to proper prototyped object